lib/modules/asset/AssetService.ts

```diff
@@ -67,34 +75,84 @@ export class AssetService {
   registerAskEvents() {
     onAsk<AskAssetRefreshModel>(
       "ask:device-manager:asset:refresh-model",
-      async ({ assetModel, assetId, engineId }) => {
-        const asset = await this.get(engineId, assetId);
+      async ({ assetModel }) => {
+        const engines = await ask<AskEngineList>(
+          "ask:device-manager:engine:list",
+          {
+            group: assetModel.engineGroup,
+          }
+        );
+
+        const targets = engines.map((engine) => ({
+          collections: [InternalCollection.ASSETS],
+          index: engine.index,
+        }));
+
+        const assetSearch = await this.sdk.query<
+          BaseRequest,
+          JSONObject // TODO: switch to DocumentSearchResult<AssetContent> once KHit<> has index and collection properties
+        >({
+          action: "search",
+          body: { query: { equals: { model: assetModel.asset.model } } },
+          controller: "document",
+          lang: "koncorde",
+          targets,
+        });
 
         const modelMetadata = {};
+
         for (const metadataName of Object.keys(
           assetModel.asset.metadataMappings
         )) {
-          modelMetadata[metadataName] = null;
-        }
-        for (const [metadataName, metadataValue] of Object.entries(
-          assetModel.asset.defaultMetadata
-        )) {
-          _.set(modelMetadata, metadataName, metadataValue);
+          const defaultMetadata =
+            assetModel.asset.defaultMetadata[metadataName];
+          modelMetadata[metadataName] = defaultMetadata ?? null;
         }
 
-        const removedMetadata = _.difference(
-          Object.keys(asset._source.metadata),
-          Object.keys(modelMetadata)
+        const removedMetadata: string[] = [];
+
+        const updatedAssetsPerIndex: {
+          [x: string]: KDocument<AssetContent>[];
+        } = assetSearch.result.hits.reduce(
+          (
+            acc: { [x: string]: KDocument<AssetContent>[] },
+            asset: JSONObject
+          ) => {
+            const assetMetadata = Object.fromEntries(
+              Object.entries(asset._source.metadata).filter(([key]) => {
+                if (!(key in modelMetadata)) {
+                  removedMetadata.push(key);
+                  return false;
+                }
+
+                return true;
+              })
+            );
```

Comment:
If the `delete` keyword is ok (in this case it seems to be)
Could you use a for loop like this:
```ts
for (const key of Object.keys(asset._source.metadata) {
  if (!(key in modelMetadata)) {
    removedMetadata.push(key);
    delete asset._source.metadata[key];
  }
}
```
suggested by @sebtiz13

Reply:
It should indeed be okay

Reply:
It should indeed be okay !